notebooks/cookbook/report_minimization.ipynb

```diff
@@ -0,0 +1,129 @@
+    "\n",
+    "The syntax for doing this is somewhat different to typical OpenMM functionality. Read the [API documentation](http://docs.openmm.org/development/api-python/generated/openmm.openmm.MinimizationReporter.html) for more explanation.\n",
+    "\n",
+    "First we will create a test system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3592a17b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openmm.app import *\n",
+    "from openmm import *\n",
+    "from openmm.unit import *\n",
+    "from sys import stdout\n",
+    "\n",
+    "pdb = PDBFile('villin.pdb')\n",
+    "forcefield = ForceField('amber14-all.xml', 'amber14/tip3pfb.xml')\n",
+    "system = forcefield.createSystem(pdb.topology, nonbondedMethod=PME,\n",
+    "        nonbondedCutoff=1*nanometer, constraints=HBonds)\n",
+    "integrator = LangevinMiddleIntegrator(300*kelvin, 1/picosecond, 0.004*picoseconds)\n",
+    "simulation = Simulation(pdb.topology, system, integrator)\n",
+    "simulation.context.setPositions(pdb.positions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e7b659d",
+   "metadata": {},
+   "source": [
+    "We then create a `MinimizationReporter` class that will print the current energy to the screen every 100 steps and save the energies to an array which we can plot later"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8692b40e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define a subclass of MinimizationReporter\n",
+    "class Reporter(MinimizationReporter):\n",
+    "    interval = 100 # report interval\n",
+    "    energies = [] # array to record progress\n",
+    "    def report(self, iteration, x, grad, args):\n",
+    "        # print current system energy to screen \n",
+    "        if iteration % self.interval == 0:\n",
+    "            print(iteration, args['system energy'])\n",
+    "\n",
+    "        # save energy at each iteration to an array we can use later\n",
+    "        self.energies.append(args['system energy'])\n",
+    "\n",
+    "        # The report method must return a bool specifying if minimization should be stopped. \n",
+    "        # You can use this functionality for early termination.\n",
+    "        return False"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9bbc621a",
+   "metadata": {},
+   "source": [
+    "We now create an instance of the reporter and minimize the system with the reporter attached."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6be58c3b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create an instance of our reporter\n",
+    "reporter = Reporter()\n",
+    "\n",
+    "# Perform local energy minimization \n",
+    "simulation.minimizeEnergy(reporter=reporter)\n",
+    "\n",
+    "# plot the minimization progress\n",
```

Comment:
Just a personal preference, but I'd remove the comments.  We don't need a separate comment for every line of code, especially when all they do is repeat what you just said immediately above: "We now create an instance of the reporter and minimize the system with the reporter attached."  I'd show the two lines that do that without additional comments, then break to a text cell that says, "Now let's plot the energies that were recorded by the reporter," and then put the plotting code in its own cell.